pkg/cmd/cli-manager/cmd.go

```diff
@@ -27,7 +28,9 @@ func NewCLIManagerCommand(name string) *cobra.Command {
 	cmd.Use = name
 	cmd.Short = "Start the CLI manager controllers"
 
-	cmd.Flags().BoolVar(&ServeArtifactAsHttp, "serve-artifacts-in-http", false, "serving artifact in HTTP instead of HTTPS. That is used for testing purposes and not recommended for production")
+	if os.Getenv(httpServeEnvVar) == "true" {
+		cmd.Flags().BoolVar(&ServeArtifactAsHttp, "serve-artifacts-in-http", false, "serving artifact in HTTP instead of HTTPS. That is used for testing purposes only. Using the flag in production is at your own risk. This flag is not supported.")
+	}
```

Comment:
`cmd.Flags().MarkHidden("serve-artifacts-in-http")` needs to go under the if branch